File: frontend/src/api/intents.ts
// 意图 API：发起意图请求、查询已声明意图。

import type { JsonValue } from "./types";
import { call } from "./core";

/** 发起意图请求（请求 / 响应式模块联动）。 */
export function intentsRequest(intent: string, payload: JsonValue): Promise<JsonValue> {
  return call<JsonValue>("intents_request", { intent, payload });
}

/** 已声明意图清单（模块名 -> 意图名）。 */
export function intentsDeclared(): Promise<Array<[string, string]>> {
  return call<Array<[string, string]>>("intents_declared");
}

File: src-tauri/src/commands/intents.rs
//! 意图命令：发起意图请求、查询已声明意图。

use serde_json::Value;
use tauri::State;

use crate::commands::into_command_error;
use crate::error::CommandResult;
use crate::state::KernelContext;

/// 发起意图请求（请求 / 响应式模块联动）。
#[tauri::command]
pub fn intents_request(
    kernel: State<'_, KernelContext>,
    intent: String,
    payload: Value,
) -> CommandResult<Value> {
    kernel
        .inner()
        .intents()
        .request(&intent, payload)
        .map_err(into_command_error)
}

/// 已声明意图清单（模块名 -> 意图名）。
#[tauri::command]
pub fn intents_declared(kernel: State<'_, KernelContext>) -> CommandResult<Vec<(String, String)>> {
    Ok(kernel.inner().intents().declared())
}
